src/catalog/catalog.cpp

```diff
@@ -1207,6 +1207,22 @@ void Catalog::Initialize(optional_ptr<ClientContext> context, bool load_builtin)
 void Catalog::FinalizeLoad(optional_ptr<ClientContext> context) {
 }
 
+void Catalog::SetEncryptionKeyId(const string &key_id) {
+	throw InternalException("SetEncryptionKeyId() is only implemented for DuckCatalog");
```

Comment:
I don't think these need to be virtual methods - can we move them to only be present in the `DuckCatalog`?